From 8e8aa80bc920b4f6bb1860a08354b22f0a324137 Mon Sep 17 00:00:00 2001
From: Dave Aldridge <fovsoft@gmail.com>
Date: Tue, 8 May 2012 13:39:09 +0100
Subject: [PATCH 39/70] pc7302/pc7308: changes for u-boot v2012.04

---
 MAKEALL  |    2 --
 Makefile |    9 ---------
 2 files changed, 0 insertions(+), 11 deletions(-)

diff --git a/MAKEALL b/MAKEALL
index 77e91a3..e5da6f1 100755
--- a/MAKEALL
+++ b/MAKEALL
@@ -321,8 +321,6 @@ LIST_SA="$(boards_by_cpu sa1100)"
 LIST_ARM9="$(boards_by_cpu arm920t)	\
 	$(boards_by_cpu arm926ejs)	\
 	$(boards_by_cpu arm925t)	\
-	picochippc7302          \
-        picochippc7308          \
 "
 
 #########################################################################
diff --git a/Makefile b/Makefile
index 075d8aa..cdd4294 100644
--- a/Makefile
+++ b/Makefile
@@ -685,15 +685,6 @@ SX1_config:		unconfig
 	fi;
 	@$(MKCONFIG) -n $@ SX1 arm arm925t sx1
 
-picochippc7302nand_config :	unconfig
-	@$(MKCONFIG) $(@:_config=) arm arm926ejs pc7302 picochip picoxcell
-
-picochippc7302nor_config :	unconfig
-	@$(MKCONFIG) $(@:_config=) arm arm926ejs pc7302 picochip picoxcell
-
-picochippc7308_config :	unconfig
-	@$(MKCONFIG) $(@:_config=) arm arm926ejs pc7308 picochip picoxcell
-
 #########################################################################
 ## ARM1176 Systems
 #########################################################################
-- 
1.7.0.4

