From 0f3979586bc77749730177bdf262aa004d56f3e9 Mon Sep 17 00:00:00 2001
From: Dave Aldridge <fovsoft@gmail.com>
Date: Mon, 29 Oct 2012 14:11:32 +0000
Subject: [PATCH 58/70] pc73032: fix some compiler warnings

---
 board/picochip/pc73032/lowlevel_init.S |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/picochip/pc73032/lowlevel_init.S b/board/picochip/pc73032/lowlevel_init.S
index d7f287a..63dd909 100644
--- a/board/picochip/pc73032/lowlevel_init.S
+++ b/board/picochip/pc73032/lowlevel_init.S
@@ -80,7 +80,7 @@
         nop
         nop
 
-        push    {r0-r14}
+        push    {r0-r12, r14}
 
         bl      initialiseBootErrorPin
 
@@ -89,7 +89,7 @@
 
         bl      pc30xx_memif_arm_init
 
-        pop {r0-r14}
+        pop     {r0-r12, r14}
 
         /* Return to my caller */
         mov	pc, lr
-- 
1.7.0.4

