From 08dc77ea5971a5cd307cdd725e65217909874e5e Mon Sep 17 00:00:00 2001
From: Jamie Iles <jamie@jamieiles.com>
Date: Mon, 7 Mar 2011 16:24:14 +0000
Subject: [PATCH 013/228] picoxcell/pc3x3: support for pin muxing on pc3x3

Signed-off-by: Jamie Iles <jamie@jamieiles.com>
---
 arch/arm/mach-picoxcell/pc3x3.c |   74 +++++++++++++++++++++++++++++++++++++++
 1 files changed, 74 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-picoxcell/pc3x3.c b/arch/arm/mach-picoxcell/pc3x3.c
index e7127c9..7fc6f4e 100644
--- a/arch/arm/mach-picoxcell/pc3x3.c
+++ b/arch/arm/mach-picoxcell/pc3x3.c
@@ -8,6 +8,7 @@
  * All enquiries to support@picochip.com
  */
 #include <linux/delay.h>
+#include <linux/gpio.h>
 #include <linux/io.h>
 #include <linux/irq.h>
 #include <linux/kernel.h>
@@ -17,6 +18,7 @@
 #include <mach/clkdev.h>
 #include <mach/hardware.h>
 
+#include "mux.h"
 #include "picoxcell_core.h"
 #include "soc.h"
 
@@ -271,6 +273,77 @@ static void pc3x3_clk_init(void)
 	__clk_disable(&otp_clk);
 }
 
+static struct mux_def pc3x3_mux[] = {
+	/*	Name		ARM	SD	PERIPH	REG	BIT	PERREG	PERBIT	FLAGS */
+	MUXGPIO(arm_gpio0,	0,	16,	RSVD,	0x34,	0,	-1,	-1,	0),
+	MUXGPIO(arm_gpio1,	1,	17,	RSVD,	0x34,	1,	-1,	-1,	0),
+	MUXGPIO(arm_gpio2,	2,	18,	RSVD,	0x34,	2,	-1,	-1,	0),
+	MUXGPIO(arm_gpio3,	3,	19,	RSVD,	0x34,	3,	-1,	-1,	0),
+	MUXGPIO(shd_gpio,	8,	8,	RSVD,	0x34,	8,	-1,	-1,	0),
+	MUXGPIO(boot_mode0,	9,	9,	RSVD,	0x34,	9,	-1,	-1,	0),
+	MUXGPIO(boot_mode1,	10,	10,	RSVD,	0x34,	10,	-1,	-1,	0),
+	MUXGPIO(sdram_speed_sel,11,	11,	RSVD,	0x34,	11,	-1,	-1,	0),
+	MUXGPIO(mii_rev_en,	12,	12,	RSVD,	0x34,	12,	-1,	-1,	0),
+	MUXGPIO(mii_rmii_en,	13,	13,	RSVD,	0x34,	13,	-1,	-1,	0),
+	MUXGPIO(mii_speed_sel,	14,	14,	RSVD,	0x34,	14,	-1,	-1,	0),
+
+	MUXGPIO(ebi_addr14,	32,	-1,	EBI,	-1,	-1,	0x3c,	0,	0),
+	MUXGPIO(ebi_addr15,	33,	-1,	EBI,	-1,	-1,	0x3c,	1,	0),
+	MUXGPIO(ebi_addr16,	34,	-1,	EBI,	-1,	-1,	0x3c,	2,	0),
+	MUXGPIO(ebi_addr17,	35,	-1,	EBI,	-1,	-1,	0x3c,	3,	0),
+	MUXGPIO(ebi_addr18,	20,	4,	EBI,	0x34,	20,	0x3c,	4,	0),
+	MUXGPIO(ebi_addr19,	21,	5,	EBI,	0x34,	21,	0x3c,	5,	0),
+	MUXGPIO(ebi_addr20,	22,	6,	EBI,	0x34,	22,	0x3c,	6,	0),
+	MUXGPIO(ebi_addr21,	23,	7,	EBI,	0x34,	23,	0x3c,	7,	0),
+	MUXGPIO(ebi_addr22,	4,	20,	EBI,	0x34,	4,	0x3c,	8,	0),
+	MUXGPIO(ebi_addr23,	5,	21,	EBI,	0x34,	5,	0x3c,	9,	0),
+	MUXGPIO(ebi_addr24,	6,	22,	EBI,	0x34,	6,	0x3c,	10,	0),
+	MUXGPIO(ebi_addr25,	7,	23,	EBI,	0x34,	7,	0x3c,	11,	0),
+	MUXGPIO(ebi_addr26,	15,	15,	EBI,	0x34,	15,	0x3c,	12,	0),
+	MUXGPIO(ebi_clk_pin,	53,	-1,	EBI,	-1,	-1,	0x3c,	13,	0),
+
+	MUXGPIO(pai_rx_data0,	20,	4,	PAI,	0x34,	20,	0x38,	8,	0),
+	MUXGPIO(pai_rx_data1,	21,	5,	PAI,	0x34,	21,	0x38,	9,	0),
+	MUXGPIO(pai_rx_data2,	22,	6,	PAI,	0x34,	22,	0x38,	10,	0),
+	MUXGPIO(pai_rx_data3,	23,	7,	PAI,	0x34,	23,	0x38,	11,	0),
+	MUXGPIO(pai_rx_data4,	28,	-1,	PAI,	-1,	-1,	0x38,	4,	0),
+	MUXGPIO(pai_rx_data5,	29,	-1,	PAI,	-1,	-1,	0x38,	5,	0),
+	MUXGPIO(pai_rx_data6,	30,	-1,	PAI,	-1,	-1,	0x38,	6,	0),
+	MUXGPIO(pai_rx_data7,	31,	-1,	PAI,	-1,	-1,	0x38,	7,	0),
+
+	MUXGPIO(pai_tx_data0,	4,	20,	PAI,	0x34,	4,	0x38,	0,	0),
+	MUXGPIO(pai_tx_data1,	5,	21,	PAI,	0x34,	5,	0x38,	1,	0),
+	MUXGPIO(pai_tx_data2,	6,	22,	PAI,	0x34,	6,	0x38,	2,	0),
+	MUXGPIO(pai_tx_data3,	7,	23,	PAI,	0x34,	7,	0x38,	3,	0),
+	MUXGPIO(pai_tx_data4,	24,	-1,	PAI,	-1,	-1,	0x38,	4,	0),
+	MUXGPIO(pai_tx_data5,	25,	-1,	PAI,	-1,	-1,	0x38,	5,	0),
+	MUXGPIO(pai_tx_data6,	26,	-1,	PAI,	-1,	-1,	0x38,	6,	0),
+	MUXGPIO(pai_tx_data7,	27,	-1,	PAI,	-1,	-1,	0x38,	7,	0),
+
+	MUXGPIO(decode0,	36,	-1,	DECODE,	-1,	-1,	0x40,	0,	0),
+	MUXGPIO(decode1,	37,	-1,	DECODE,	-1,	-1,	0x40,	1,	0),
+	MUXGPIO(decode2,	38,	-1,	DECODE,	-1,	-1,	0x40,	2,	0),
+	MUXGPIO(decode3,	39,	-1,	DECODE,	-1,	-1,	0x40,	3,	0),
+
+	MUXGPIO(ssi_clk,	40,	-1,	SSI,	-1,	-1,	0x44,	0,	0),
+	MUXGPIO(ssi_data_in,	41,	-1,	SSI,	-1,	-1,	0x44,	0,	0),
+	MUXGPIO(ssi_data_out,	42,	-1,	SSI,	-1,	-1,	0x44,	0,	0),
+
+	MUXGPIO(mii_tx_data2,	43,	-1,	MII,	-1,	-1,	0,	13,	MUX_RO),
+	MUXGPIO(mii_tx_data3,	44,	-1,	MII,	-1,	-1,	0,	13,	MUX_RO),
+	MUXGPIO(mii_rx_data2,	45,	-1,	MII,	-1,	-1,	0,	13,	MUX_RO),
+	MUXGPIO(mii_rx_data3,	46,	-1,	MII,	-1,	-1,	0,	13,	MUX_RO),
+	MUXGPIO(mii_col,	47,	-1,	MII,	-1,	-1,	0,	13,	MUX_RO),
+	MUXGPIO(mii_crs,	48,	-1,	MII,	-1,	-1,	0,	13,	MUX_RO),
+	MUXGPIO(mii_tx_clk,	49,	-1,	MII,	-1,	-1,	0,	13,	MUX_RO),
+
+	MUXGPIO(max_tx_ctrl,	50,	-1,	MAXIM,	-1,	-1,	0x44,	1,	0),
+	MUXGPIO(max_ref_clk,	51,	-1,	MAXIM,	-1,	-1,	0x44,	1,	0),
+	MUXGPIO(max_trig_clk,	52,	-1,	MAXIM,	-1,	-1,	0x44,	1,	0),
+
+	MUXGPIO(sdgpio0,	-1,	0,	FRACN,	-1,	-1,	0,	7,	MUX_INVERT_PERIPH),
+};
+
 static const struct picoxcell_timer pc3x3_timers[] = {
 	{
 		.name	= "timer0",
@@ -315,4 +388,5 @@ struct picoxcell_soc pc3x3_soc = {
 
 static void pc3x3_init(void)
 {
+	picoxcell_mux_register(pc3x3_mux, ARRAY_SIZE(pc3x3_mux));
 }
-- 
1.7.0.4

