From c9237c4ca1388b6a72e017a62c933dc8603a5640 Mon Sep 17 00:00:00 2001
From: Jamie Iles <jamie@jamieiles.com>
Date: Mon, 11 Apr 2011 15:15:04 +0100
Subject: [PATCH 067/228] pc3x2: sdgpio support

Signed-off-by: Jamie Iles <jamie@jamieiles.com>
---
 arch/arm/mach-picoxcell/pc3x2.c |   44 +++++++++++++++++++++++++++++++++++++++
 1 files changed, 44 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-picoxcell/pc3x2.c b/arch/arm/mach-picoxcell/pc3x2.c
index dadc081..bedc15b 100644
--- a/arch/arm/mach-picoxcell/pc3x2.c
+++ b/arch/arm/mach-picoxcell/pc3x2.c
@@ -115,10 +115,54 @@ struct picoxcell_soc pc3x2_soc = {
 	.nr_timers	= ARRAY_SIZE(pc3x2_timers),
 };
 
+static const char * const pc3x2_sdgpio_lo_pins[] = {
+	"sdgpio0",
+	"sdgpio1",
+	"sdgpio2",
+	"sdgpio3",
+	"sdgpio4",
+	"sdgpio5",
+	"sdgpio6",
+	"sdgpio7",
+};
+
+static const char * const pc3x2_sdgpio_shared_pins[] = {
+	"sdgpio8",
+	"sdgpio9",
+	"sdgpio10",
+	"sdgpio11",
+	"sdgpio12",
+	"sdgpio13",
+	"sdgpio14",
+	"sdgpio15",
+};
+
+static const struct sdgpio_platform_data pc3x2_sdgpio = {
+	.banks				= {
+		{
+			.names		= pc3x2_sdgpio_lo_pins,
+			.block_base	= 0,
+			.gpio_start	= PC3X2_GPIO_PIN_SDGPIO_0,
+			.nr_pins	= ARRAY_SIZE(pc3x2_sdgpio_lo_pins),
+			.label		= "sdpio_lo",
+		},
+		{
+			.names		= pc3x2_sdgpio_shared_pins,
+			.block_base	= 8,
+			.gpio_start	= PC3X2_GPIO_PIN_SDGPIO_8,
+			.nr_pins	= ARRAY_SIZE(pc3x2_sdgpio_shared_pins),
+			.label		= "sdgpio_shared",
+		},
+	},
+	.nr_banks			= 2,
+};
+
 static void pc3x2_add_gpio(void)
 {
 	picoxcell_add_gpio_port(0, 8, PC3X2_GPIO_PIN_ARM_0);
 	picoxcell_add_gpio_port(1, 8, PC3X2_GPIO_PIN_ARM_8);
+	platform_device_register_data(NULL, "sdgpio", -1, &pc3x2_sdgpio,
+		sizeof(pc3x2_sdgpio));
 }
 
 static void pc3x2_init_bus_snoopers(void)
-- 
1.7.0.4

