From 340a93dd1a38f8202cf2d72a7d00fc0127e6b361 Mon Sep 17 00:00:00 2001
From: Andrew Watkins <Andrew.Watkins@picochip.com>
Date: Wed, 2 Nov 2011 10:40:38 +0000
Subject: [PATCH 182/228] picoif: Fixed the cause of the .pico load file fail

---
 drivers/picochip/picoif_main.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/picochip/picoif_main.c b/drivers/picochip/picoif_main.c
index 82ce5ac..6be1f36 100644
--- a/drivers/picochip/picoif_main.c
+++ b/drivers/picochip/picoif_main.c
@@ -863,12 +863,12 @@ picoif_ioctl_multi_reg_req( struct picoif_multi_reg_req __user *user_req,
         {
             sg_init_one( &sgl, bounce,
                          min( ( multi_reg_req.count - words_so_far ) *
-                         sizeof( u32 ), PICOIF_MAX_TRANSFER_SIZE / sizeof( u32 ) ) );
+                         sizeof( u32 ), PICOIF_MAX_TRANSFER_SIZE ) );
             sg_mark_end( &sgl );
 
             ret = copy_from_user( bounce, &multi_reg_req.buf[ words_so_far ],
                           sgl.length * sizeof( u32 ));
-            words_so_far += sgl.length;
+            words_so_far += sgl.length / sizeof( u32 );
 
             if ( ret )
                 goto out;
-- 
1.7.0.4

